packages/core/src/node/console-logger-server.ts

```diff
@@ -59,7 +64,22 @@ export class ConsoleLoggerServer implements ILoggerServer {
     async log(name: string, logLevel: number, message: string, params: any[]): Promise<void> {
         const configuredLogLevel = await this.getLogLevel(name);
         if (logLevel >= configuredLogLevel) {
-            ConsoleLogger.log(name, logLevel, message, params);
+            const fullMessage = ConsoleLogger.log(name, logLevel, message, params);
+            this.logToFile(fullMessage, params);
+        }
+    }
+
+    protected logToFile(message: string, params: any[]): void {
+        if (this.cli.logFile && !this.logFileStream) {
+            this.logFileStream = fs.createWriteStream(this.cli.logFile, { flags: 'a' });
+            // Only log errors once to avoid spamming the console
+            this.logFileStream.once('error', error => {
+                console.error(`Error writing to log file ${this.cli.logFile}`, error);
```

Comment:
If the log file creation fails for whatever reason, both `this.cli.logFile` and `this.logFileStream` will be falsy, and the error message about the log file write failure will never be visible.

<img width="744" alt="Screenshot 2025-02-11 at 08 23 04" src="https://github.com/user-attachments/assets/26426c6c-0678-4060-9813-4bf04d713607" />

_I made the log file creation fail by providing an invalid filename (`--log-file=&&&\\//.xxx`), but the file creation could fail due to another error, for example, `EPERM`._

Reply:
Closing as invalid. It's literally two entries above that the log file creation has failed.